pxtcompiler/emitter/decompiler.ts

```diff
@@ -2369,7 +2421,22 @@ ${output}</xml>`;
                 }
             }
 
-            eventStatements.map(n => getStatementBlock(n, undefined, undefined, false, topLevel)).forEach(emitStatementNode);
+            const compiledEvents: StatementNode[] = [];
+            const keyMap: {[index: string]: boolean} = {};
+            for (const statement of eventStatements) {
```

Comment:
In theory this should be going through the statements in the order of the program, right? If so, then we should be able to say that since these would get dropped in on start we'll typically maintain the correct behavior for repeated event handlers from javascript -> blocks most of the time right (/ the correct event will usually be set at the end)?

(except I guess if the first event was intentionally registered after other things in on start for some reason, or if `afterOnStart=true` is set on the block.)